skops/io/_persist.py

```diff
@@ -133,14 +144,64 @@ def loads(data):
     Parameters
     ----------
     data: bytes
-        The file name of the object to be loaded.
+        The dumped data to be loaded in bytes format.
+
+    trusted: bool, or list of str, default=False
+        If ``True``, the object will be loaded without any security checks. If
+        ``False``, the object will be loaded only if there are only trusted
+        objects in the dumped file. If a list of strings, the object will be
+        loaded only if there are only trusted objects and objects of types
+        listed in ``trusted`` are in the dumped file.
 
+    Returns
+    -------
+    instance: object
+        The loaded object.
     """
     if isinstance(data, str):
         raise TypeError("Can't load skops format from string, pass bytes")
 
-    with ZipFile(io.BytesIO(data), "r") as input_zip:
-        schema = json.loads(input_zip.read("schema.json"))
-        load_context = LoadContext(src=input_zip)
-        instance = get_instance(schema, load_context=load_context)
+    with ZipFile(io.BytesIO(data), "r") as zip_file:
+        schema = json.loads(zip_file.read("schema.json"))
+        load_context = LoadContext(src=zip_file)
+        tree = get_tree(schema, load_context)
+        audit_tree(tree, trusted)
+        instance = tree.construct()
+
     return instance
+
+
+def get_untrusted_types(*, data=None, file=None):
+    """Get a list of untrusted types in a skops dump.
+
+    Parameters
+    ----------
+    data: bytes
+        The data to be checked, in bytes format.
+
+    file: str or Path
+        The file to be checked.
+
+    Returns
+    -------
+    untrusted_types: list of str
+        The list of untrusted types in the dump.
+
+    Notes
+    -----
+    Only one of data or file should be passed.
+    """
+    if data and file:
+        raise ValueError("Only one of data or file should be passed.")
+
+    if data:
+        content = io.BytesIO(data)
+    else:
+        content = file
+
+    with ZipFile(content, "r") as zip_file:
+        schema = json.loads(zip_file.read("schema.json"))
+        tree = get_tree(schema, load_context=LoadContext(src=zip_file))
+        untrusted_types = tree.get_unsafe_set()
+
+    return list(untrusted_types)
```

Comment:
Similar argument as above, do we want to return `sorted(untrusted_types)` here?

Reply:
It's a weird quirk that dicts have keys sorted by insert order since Python 3.7, but sets still don't.

Example:

```
foo = {1, 10, 5}
list(foo) # {1, 10, 5}

foo.add(3)
list(foo)  # {1, 10, 3, 5}
```